### PR TITLE
Fixes #4405: Renaming event logger to analytics event logger

### LINKED ISSUE #4405: Renaming EventLogger to AnalyticsEventLogger
Since we're moving towards more specific event loggers like the PerformanceMetricsEventLogger, the existing EventLogger has become too generic and hence, it must be refactored. 

Potential checklist: 
- [ ] changes in all the production files where eventLogger is used,
- [ ] changes in test files.
- [ ] changes in dagger modules.
- [ ] refactoring of fakeEventlogger to fakeAnalyticsEventLogger

### PR DESCRIPTION
**Pull request by Akshatkamboj14**
## Explanation
Fixes #4405 

## Potential checklist:
- [x]  changes in all the production files where eventLogger is used,
- [x]  changes in test files.
- [x]  changes in dagger modules.
- [x]  refactoring of fakeEventlogger to fakeAnalyticsEventLogger

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/devoptions/vieweventlogs/ViewEventLogsViewModel.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import javax.inject.Inject
  */
 @FragmentScope
 class ViewEventLogsViewModel @Inject constructor(
-  debugEventLogger: DebugEventLogger,
+  debugEventLogger: DebugAnalyticsEventLogger,
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/devoptions/vieweventlogs/ViewEventLogsViewModel.kt]: Rename `debugEventLogger ` to `debugAnalyticsEventLogger`.

Same applies for all other files too.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Akshatkamboj14 PTAL. Thanks

File: app/src/test/java/org/oppia/android/app/player/exploration/ExplorationActivityLocalTest.kt [hunk 2]
```diff
@@ -100,7 +100,7 @@ class ExplorationActivityLocalTest {
   val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
 
   @Inject
-  lateinit var fakeEventLogger: FakeEventLogger
+  lateinit var fakeEventLogger: FakeAnalyticsEventLogger
```
Comment (BenHenning) [inline app/src/test/java/org/oppia/android/app/player/exploration/ExplorationActivityLocalTest.kt]: Here & elsewhere: I suggest updating parameters and fields to be either ``fakeAnalyticsEventLogger`` or ``fakeAnalyticsLogger`` to capture the context (and ditto for parameters/variables of DebugAnallyticsEventLogger and AnalyticsEventLogger).

File: utility/src/main/java/org/oppia/android/util/logging/firebase/LogReportingModule.kt [hunk 3]
```diff
@@ -17,5 +17,7 @@ class LogReportingModule {
 
   @Provides
   @Singleton
-  fun provideEventLogger(factory: FirebaseEventLogger.Factory): EventLogger = factory.create()
+  fun provideEventLogger(
```
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/logging/firebase/LogReportingModule.kt]:
```suggestion
  fun provideFirebaseAnalyticsEventLogger(
```

Per my earlier comment.

**Review by BenHenning — APPROVED** [review]
Thanks @Akshatkamboj14! The PR generally LGTM, but besides a couple of comments I think that you also need to:
- Make sure there are no TODOs in code for #4405 that need to be removed (otherwise remove them)
- Finish the PR description & make sure that it's fully filled out correctly (look at other PRs if you need a reference for how to fill it out for non-user facing PRs)
- Fix the broken CI checks/tests

File: app/src/main/java/org/oppia/android/app/devoptions/vieweventlogs/ViewEventLogsViewModel.kt (hunk 1 above)
Comment (Akshatkamboj14) [reply app/src/main/java/org/oppia/android/app/devoptions/vieweventlogs/ViewEventLogsViewModel.kt]: done

File: app/src/test/java/org/oppia/android/app/player/exploration/ExplorationActivityLocalTest.kt (hunk 2 above)
Comment (Akshatkamboj14) [reply app/src/test/java/org/oppia/android/app/player/exploration/ExplorationActivityLocalTest.kt]: done.

File: utility/src/main/java/org/oppia/android/util/logging/firebase/LogReportingModule.kt (hunk 3 above)
Comment (Akshatkamboj14) [reply utility/src/main/java/org/oppia/android/util/logging/firebase/LogReportingModule.kt]: @BenHenning can you send me the link of some PRs so I can take a reference from like how to fill the description of non-user facing PRs?

Comment (seanlip) [reply utility/src/main/java/org/oppia/android/util/logging/firebase/LogReportingModule.kt]: @Akshatkamboj14 This is something that you can look for directly. See https://github.com/oppia/oppia-android/pulls.

**Review by rt4914 — APPROVED** [review]
Approved for code-owners

**Review by BenHenning — CHANGES_REQUESTED** [review]
Took another pass. I actually meant to request changes last time since CI checks are still breaking, so updating my review status.

**Comment by Akshatkamboj14** [comment]
@BenHenning IDK why my checks are failing. can you help me to figure it out?

**Comment by BenHenning** [comment]
> @BenHenning IDK why my checks are failing. can you help me to figure it out?

@Akshatkamboj14 apologies I missed this (in the future, please assign me or others if we need to take action on the PR).

Regarding the errors, what specific error are you stuck on, what have you tried, and what do you think might be wrong? This is a good situation to consider writing up a brief debugging doc (https://github.com/oppia/oppia/wiki/Debugging-Docs) and sharing it to provide more specifics--these can be really helpful to help with investigation.

**Pull request closed without merging**